src/ssmDocument/util/util.ts

```diff
@@ -0,0 +1,29 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as path from 'path'
+import * as vscode from 'vscode'
+import * as fs from 'fs-extra'
+
+export async function openAndSaveDocument(
+    content: string,
+    filename: string,
+    language: string
+): Promise<vscode.TextDocument> {
+    const wsPath = vscode.workspace.workspaceFolders ? vscode.workspace.workspaceFolders[0].uri.fsPath : '/'
+    let filePath = path.join(wsPath, filename)
+    const fileInfo = await vscode.window.showSaveDialog({ defaultUri: vscode.Uri.file(filePath) })
+
+    if (fileInfo) {
+        filePath = fileInfo.fsPath
+        fs.writeFileSync(filePath, content)
+        const openPath = vscode.Uri.file(filePath)
+
+        return await vscode.workspace.openTextDocument(openPath)
+    }
+
+    // The user didn't save the file, so just open an untitiled file
```

Comment:
Is there value in having the toolkit save the file? We've been providing guidance to put new content like this into a new untitled document. You can still set the document's language, it simplifies the toolkit code, and users can choose to save the file wherever they like, or discard the document.

Reply:
I want to prompt the users to save the document as {doc name}.{doc type}.ssm.json, so that additional validation on document type and schemaVersion can be performed.